Makefile

```
@@ -274,7 +291,19 @@ DEP_FILES := $(O_FILES:.o=.asmproc.d) $(OVL_RELOC_FILES:.o=.d)
 OTHER_DIRS := baserom dmadata linker_scripts
 
 # create build directories
-$(shell mkdir -p $(foreach dir,$(SRC_DIRS) $(ASM_DIRS) $(ASSET_BIN_DIRS) $(ASSET_BIN_DIRS_C_FILES) $(OTHER_DIRS),$(BUILD_DIR)/$(dir)))
+$(shell mkdir -p $(foreach dir,$(OTHER_DIRS),$(BUILD_DIR)/$(dir)) \
+                 $(foreach dir, \
+                     $(SRC_DIRS) \
+                     $(ASM_DIRS) \
+                     $(ASSET_BIN_DIRS) \
+                     $(ASSET_BIN_DIRS_C_FILES) \
+                     $(SAMPLE_DIRS) \
+                     $(SAMPLEBANK_DIRS),\
+                   $(BUILD_DIR)/$(dir)) \
+                 $(foreach dir,\
+                     $(SAMPLE_EXTRACT_DIRS) \
+                     $(SAMPLEBANK_EXTRACT_DIRS), \
+                   $(dir:$(EXTRACTED_DIR)/%=$(BUILD_DIR)/%)))
```

```suggestion
$(shell mkdir -p $(foreach dir,\
                     $(SRC_DIRS) \
                     $(ASM_DIRS) \
                     $(OTHER_DIRS),\
                     $(BUILD_DIR)/$(dir)))
$(shell mkdir -p $(foreach dir,\
                     $(ASSET_BIN_DIRS) \
                     $(ASSET_BIN_DIRS_C_FILES) \
                     $(SAMPLE_DIRS) \
                     $(SAMPLEBANK_DIRS),\
                   $(BUILD_DIR)/$(dir)) \
                 $(foreach dir,\
                     $(SAMPLE_EXTRACT_DIRS) \
                     $(SAMPLEBANK_EXTRACT_DIRS),\
                   $(dir:$(EXTRACTED_DIR)/%=$(BUILD_DIR)/%)))
```
Looks MM specific, but at least for me, the `mkdir` argument list has gotten too long
```bash
make: /bin/bash: Argument list too long
```
My suggestion is to break it up into 2 commands. The way I have it up above splits it into roughly src directories in one command and assets in the other.

---

I've tried splitting it into 3 commands, could you check again?

---

Seems to work for me now. Thank you :)